src/aws/auth.rs

```diff
@@ -184,6 +269,43 @@ mod tests {
         assert!(matches!(config.auth, AwsAuthentication::Role { .. }));
     }
 
+    #[test]
+    fn parsing_assume_role_with_imds_client() {
+        let config = toml::from_str::<ComponentConfig>(
+            r#"
+            auth.assume_role = "root"
+            auth.imds.max_attempts = 5
+            auth.imds.connect_timeout = 30
+            auth.imds.read_timeout = 10
+        "#,
+        )
+        .unwrap();
+
+        const CONNECT_TIMEOUT: Duration = Duration::from_secs(30);
+        const READ_TIMEOUT: Duration = Duration::from_secs(10);
```

Comment:
Just as a nit: since these are identical to the consts on lines 232-233, you might want to promote them up to the top of the `tests` module for reuse.